assets/js/blocks/collection-filters/inner-blocks/rating-filter/block.tsx

```diff
@@ -0,0 +1,321 @@
+								)
+								.map( ( option ) => option.value ) }
+							disabled={ isLoading }
+							placeholder={ __(
+								'Select Rating',
+								'woo-gutenberg-products-block'
+							) }
+							onChange={ () => {
+								// noop
+							} }
+							value={ checked }
+							// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+							// @ts-ignore - FormTokenField doesn't accept custom components, forcing it here to display component
+							displayTransform={ ( value ) => {
+								const resultWithZeroCount = {
+									value,
+									label: (
+										<Rating
+											key={
+												Number( value ) as RatingValues
+											}
+											rating={
+												Number( value ) as RatingValues
+											}
+											ratedProductsCount={ 0 }
+										/>
+									),
+								};
+								const resultWithNonZeroCount =
+									displayedOptions.find(
+										( option ) => option.value === value
+									);
+
+								const displayedResult =
+									resultWithNonZeroCount ||
+									resultWithZeroCount;
+
+								const { label, value: rawValue } =
+									displayedResult;
+
+								// A label - JSX component - is extended with faked string methods to allow using JSX element as an option in FormTokenField
+								const extendedLabel = Object.assign(
+									{},
+									label,
+									{
+										toLocaleLowerCase: () => rawValue,
+										substring: (
+											start: number,
+											end: number
+										) =>
+											start === 0 && end === 1
+												? label
+												: '',
+									}
+								);
+								return extendedLabel;
+							} }
+							saveTransform={ formatSlug }
+							messages={ {
+								added: __(
+									'Rating filter added.',
+									'woo-gutenberg-products-block'
+								),
+								removed: __(
+									'Rating filter removed.',
+									'woo-gutenberg-products-block'
+								),
+								remove: __(
+									'Remove rating filter.',
+									'woo-gutenberg-products-block'
+								),
+								__experimentalInvalid: __(
+									'Invalid rating filter.',
+									'woo-gutenberg-products-block'
+								),
+							} }
+						/>
+						{ showChevron && (
+							<Icon icon={ chevronDown } size={ 30 } />
+						) }
+					</>
+				) : (
+					<CheckboxList
+						className={ 'wc-block-rating-filter-list' }
+						options={ displayedOptions }
+						checked={ checked }
+						onChange={ () => {
+							// noop
+						} }
+						isLoading={ isLoading }
+						isDisabled={ isDisabled }
+					/>
+				) }
+			</div>
+			{
+				<div className="wc-block-rating-filter__actions">
+					{ ( checked.length > 0 || isEditor ) && ! isLoading && (
+						<FilterResetButton
+							onClick={ () => {
+								setChecked( [] );
+								setProductRatingsQuery( [] );
+								// onSubmit( [] );
+							} }
+							screenReaderLabel={ __(
+								'Reset rating filter',
+								'woo-gutenberg-products-block'
+							) }
+						/>
+					) }
```

Comment:
We also haven't implement the reset feature as well. We should have an unified solution for all.